mllib/src/main/scala/org/apache/spark/mllib/optimization/GradientDescent.scala

```diff
@@ -178,7 +178,7 @@ object GradientDescent extends Logging {
       regParam: Double,
       miniBatchFraction: Double,
       initialWeights: Vector,
-      convergenceTol: Double): (Vector, Array[Double]) = {
+      convergenceTol: Double): (Vector, Array[Double], Integer) = {
```

Comment:
Do you have to change the API here?  The loss history should have length = num iterations, right?